views.py

```diff
@@ -400,3 +400,31 @@ def add_article(request):
     return render(request, template, context)
 
 
+@has_journal
+@editor_user_required
+def discount_apc(request, apc_id):
+    """
+    Allows editor to discount an APC.
+    """
+    apc = get_object_or_404(
+        models.ArticleAPC,
+        pk=apc_id,
+        article__journal=request.journal,
+    )
+
+    if request.POST:
+        new_apc_amount = request.POST.get('new_value')
+        apc.value = new_apc_amount
+        apc.save()
```

Comment:
We are not capturing the fact that a discount is applied, I'd suggest adding a LogEntry record so that the change can be seen form the article history log, as well as capturing the actor and the discount name when one of the persisted discounts  is used